components/camel-http/src/main/java/org/apache/camel/component/http/HttpComponent.java

```diff
@@ -312,6 +312,26 @@ public Producer createProducer(CamelContext camelContext, String host,
         if (!ObjectHelper.isEmpty(uriTemplate)) {
             url += "/" + uriTemplate;
         }
+        
+        RestConfiguration config = configuration;
```

Comment:
I would suggest a default method in an interface and implementing in all components to avoid similar code piece in every component.